src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas1/Cas1PremisesController.kt

```diff
@@ -85,7 +91,42 @@ class Cas1PremisesController(
     bookingsCriteriaFilter: List<Cas1SpaceBookingCharacteristic>?,
     bookingsSortDirection: SortDirection?,
     bookingsSortBy: Cas1SpaceBookingDaySummarySortField?,
-  ): ResponseEntity<Cas1PremiseDaySummary> {
-    return super.getDaySummary(premisesId, date, bookingsCriteriaFilter, bookingsSortDirection, bookingsSortBy)
+  ): ResponseEntity<Cas1PremisesDaySummary> {
+    userAccessService.ensureCurrentUserHasPermission(UserPermission.CAS1_PREMISES_VIEW)
+
+    val premiseSummary = cas1PremisesService.getPremisesSummary(premisesId)
+    val premiseSummaryInfo = extractEntityFromCasResult(premiseSummary)
+    val premiseCapacity = cas1PremisesService.getPremiseCapacity(premisesId, date, date)
+    val premisesCapacitySummary = cas1PremiseCapacityTransformer.toCas1PremiseCapacitySummary(
+      premiseSummaryInfo = premiseSummaryInfo,
+      premiseCapacity = extractEntityFromCasResult(premiseCapacity),
+    )
```

Comment:
it may be worth embedding all the lines above into a nested set of calls, reducing the number of variables?